From ce7c53c7e063463632df9b1f392a319413025373 Mon Sep 17 00:00:00 2001
From: Andrew Boie <andrew.p.boie@intel.com>
Date: Tue, 19 Mar 2013 11:30:48 -0700
Subject: always try to create by-name links based on install id

Previously, if a block device was detected as a platform device,
the by-name logic which uses install id was skipped in favor of
the older logic which includes the bus in the path name. This
resulted in the installation of Android failing on these devices
since the expected devices nodes don't show up after the disk is
partitioned.

Now always try to use the install id logic, and fall back to the
older mechanism only if it doesn't produce any symlinks.

Change-Id: Idc096eb6425af478ddca7fe4ba9c0fe24b922d6e
Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>
---
 init/devices.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/init/devices.c b/init/devices.c
index f9f3c65..fd55c81 100644
--- a/init/devices.c
+++ b/init/devices.c
@@ -572,7 +572,8 @@ static void handle_block_device_event(struct uevent *uevent)
     snprintf(devpath, sizeof(devpath), "%s%s", base, name);
     make_dir(base, 0755);
 
-    if (!strncmp(uevent->path, "/devices/", 9))
+    links = parse_gpt_block_device(uevent);
+    if (!links && !strncmp(uevent->path, "/devices/", 9))
         links = parse_platform_block_device(uevent);
     else
         links = parse_gpt_block_device(uevent);
-- 
1.7.4.1

